homeassistant/components/nordpool/services.py

```diff
@@ -0,0 +1,127 @@
+"""Services for Nord Pool integration."""
+
+from __future__ import annotations
+
+from datetime import date, datetime
+import logging
+from typing import TYPE_CHECKING
+
+from pynordpool import (
+    Currency,
+    NordPoolAuthenticationError,
+    NordPoolEmptyResponseError,
+    NordPoolError,
+)
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import ATTR_DATE
+from homeassistant.core import (
+    HomeAssistant,
+    ServiceCall,
+    ServiceResponse,
+    SupportsResponse,
+)
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.selector import ConfigEntrySelector
+from homeassistant.util import dt as dt_util
+from homeassistant.util.json import JsonValueType
+
+if TYPE_CHECKING:
+    from . import NordPoolConfigEntry
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+ATTR_CONFIG_ENTRY = "config_entry"
+ATTR_AREAS = "areas"
+ATTR_CURRENCY = "currency"
+
+SERVICE_GET_PRICES_FOR_DATE = "get_prices_for_date"
+SERVICE_GET_PRICES_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_CONFIG_ENTRY): ConfigEntrySelector(),
+        vol.Required(ATTR_DATE): cv.date,
+        vol.Optional(ATTR_AREAS, default=[]): cv.ensure_list,
+        vol.Optional(ATTR_CURRENCY): cv.string,
+    }
+)
+
+
+def get_config_entry(hass: HomeAssistant, entry_id: str) -> NordPoolConfigEntry:
+    """Return config entry."""
+    if not (entry := hass.config_entries.async_get_entry(entry_id)):
+        raise ServiceValidationError(
+            translation_domain=DOMAIN,
+            translation_key="entry_not_found",
+        )
+    if entry.state is not ConfigEntryState.LOADED:
+        raise ServiceValidationError(
+            translation_domain=DOMAIN,
+            translation_key="entry_not_loaded",
+        )
+    return entry
+
+
+def async_setup_services(hass: HomeAssistant) -> None:
+    """Set up services for Nord Pool integration."""
+
+    async def get_prices_for_date(call: ServiceCall) -> ServiceResponse:
+        """Get price service."""
+        entry = get_config_entry(hass, call.data[ATTR_CONFIG_ENTRY])
+        asked_date: date = call.data[ATTR_DATE]
+        client = entry.runtime_data.client
+        areas: list[str] | None = call.data.get(ATTR_AREAS)
```

Comment:
`default=[]`, can we expect the areas to always be filled here? Not sure if default means that it will always be sent

Reply:
I would think most are going to leave area and currency empty to just get it from the config entry data instead (which is the default if not provided).

Reply:
Oh I know, but we have a default, can it be `None` here?

Reply:
I changed it now so it is never `None` anymore

Reply:
Crap. My changes wasn't pushed. Will fix

Reply:
Now everything should be sorted, no idea what happened before with my changes.